docs/csharp/advanced-topics/interop/walkthrough-office-programming.md

```diff
@@ -0,0 +1,131 @@
+---
+title: "Walkthrough: Office Programming - C#"
+description: Learn about the features Visual Studio offers that improve Microsoft Office programming. 
+ms.date: 02/15/2023
+ms.topic: tutorial
+dev_langs:
+  - "csharp"
+helpviewer_keywords:
+  - "Office programming [C#]"
+---
+# Walkthrough: Office Programming in C\#
+
+C# offers features that improve Microsoft Office programming. Helpful C# features include named and optional arguments and return values of type `dynamic`. In COM programming, you can omit the `ref` keyword and gain access to indexed properties.
+
+Both languages enable embedding of type information, which allows deployment of assemblies that interact with COM components without deploying primary interop assemblies (PIAs) to the user's computer. For more information, see [Walkthrough: Embedding Types from Managed Assemblies](../../../standard/assembly/embed-types-visual-studio.md).
+
+This walkthrough demonstrates these features in the context of Office programming, but many of these features are also useful in general programming. In the walkthrough, you use an Excel Add-in application to create an Excel workbook. Next, you create a Word document that contains a link to the workbook. Finally, you see how to enable and disable the PIA dependency.
+
+## Prerequisites
+
+You must have Microsoft Office Excel and Microsoft Office Word installed on your computer to complete this walkthrough.
+
+[!INCLUDE[note_settings_general](~/includes/note-settings-general-md.md)]
+
+### To set up an Excel Add-in application
```

Comment:
I think you could easily trim these headers (and the ones in the VB article) to ## and get them in the TOC. Right now the TOC is simply:

- Prerequisites
- See Also